pkg/forwarder/domain_forwarder.go

```diff
@@ -109,14 +112,25 @@ func (f *domainForwarder) retryTransactions(retryBefore time.Time) {
 				transactionsDropped.Add(1)
 				tlmTxDropped.Inc(f.domain)
 			}
-		} else if len(newQueue) < f.retryQueueLimit {
-			newQueue = append(newQueue, t)
-			transactionsRequeued.Add(1)
-			tlmTxRequeud.Inc(f.domain)
 		} else {
-			droppedRetryQueueFull++
-			transactionsDropped.Add(1)
-			tlmTxDropped.Inc(f.domain)
+			var retry bool
+			newTotalPayloadsSize := totalPayloadsSize + t.GetPayloadSize()
+			if f.retryQueueAllPayloadsMaxSize > 0 {
+				retry = newTotalPayloadsSize <= f.retryQueueAllPayloadsMaxSize
+			} else {
+				retry = len(newQueue) < f.retryQueueLimit
+			}
+
+			if retry {
+				newQueue = append(newQueue, t)
+				transactionsRequeued.Add(1)
+				tlmTxRequeud.Inc(f.domain)
+				totalPayloadsSize = newTotalPayloadsSize
```

Comment:
Since we now have this information we should export it in expvar, it would be super useful to troubleshoot memory issue with the agent. Do we also want to display this in the status page ?

Reply:
I think it would be better to do this change in https://github.com/DataDog/datadog-agent/pull/6491 in order to avoid conflict. @bmikaili What do you think?

Reply:
I agree. I'll add it once you merge this.